### ISSUE TITLE
Possible tiny improvement

### ISSUE THREAD

**Issue by ceraolo** [PowerGrids/PowerGrids#166]
During a cross check that a PhD student of mine is doing between DigSilent and PowerGrids, we considered whether parameters are better expressed in components' masks in PU or in SI units.
PowerGrids does as follows:
- generators in PU
- transformers in ohm and siemens
- lines in ohm.

Digisilent does as follows:
- generators in PU
- transformers PU 
- lines in ohm/km (and probalby siemens/km, not sure)

I am rather agnostic for lines.

Regarding transformers, I think it is better if PowerGrids adapts to DigSilent's way, both because of homogeneity with what is done for synchronous machines, and because parameters expressed in PU of machine basis are much more significant for engineers than in ohms

**Comment by casella** [comment]
I guess we could have both, with a parameter puParameters to select which ones should be considered.

**Comment by ceraolo** [comment]
Are you sure? Not always adding choice enhances usability. For Synchrnous machines I don't think anybody would use ohms in masks.  I'm afraid this is the same for transformers.
We are thinking of power engineers for which a PU reactance of 10% means something, a reactance of 3.55 ohm nothing.
Note that this could also be the root cause of #167.

**Comment by casella** [comment]
We can discuss this with @joyelfeghali and @marcochiaramello. It's a non-backwards compatible change, so we need to be careful.

**Comment by joyelfeghali** [comment]
> We can discuss this with @joyelfeghali and @marcochiaramello. It's a non-backwards compatible change, so we need to be careful.

I agree with having the parameters value of the transformer in pu. If we want also to keep the values in SI units it could be defined as a final parameter, for example ?

**Comment by casella** [comment]
> > We can discuss this with @joyelfeghali and @marcochiaramello. It's a non-backwards compatible change, so we need to be careful.
> 
> I agree with having the parameters value of the transformer in pu. If we want also to keep the values in SI units it could be defined as a final parameter, for example ?

Absolutely. The transformer models are variants of the PI network base class, which is a physical model with currents and voltages in ampere and volts, not in p.u. So, we can make the current parameters final and add new ones in p.u. to parameterize the transformer model.

In fact, this could even allow to put reasonable default values for the p.u. impedance, if their variability in real life is limited. What do you think?

**Comment by ceraolo** [comment]
> In fact, this could even allow to put reasonable default values for the p.u. impedance, if their variability in real life is limited. What do you think?

Transformer values in PU are very coherent and rich of information.
Larger Xcc's mean lower short circuits currents, but larger voltage drops. The compromise is around 4-6% for LV/MV transformers, up to 15-20% for very large AAT units.
The Xcc/Rcc ratio is typically very large, over 100; the larger this value the more efficient the transformer. 
So, if users input transformer data un PU, they know what they are doing, and this prevents mistakes, such as the (probable, I still wait confirmation from someone that knows about it) mentioned in #167.

I would not recommend a default for these values. Even though they are very meaningful and do not vary too much, there is still a huge difference between a transformer with 4% and 20% Xcc.

**Comment by AndreaBartolini** [comment]
What should be the base for the p.u. impedance? I suppose something like this:

Zbase = (Unom)^2/Snom 

but what Unom (port_a or port_b) we have to use?

Inside the `transformer` model is stated that:
```modelica
SNom = UNomB^2/CM.abs(Complex(R,X))
```
So it seems that the base for the p.u. should be

Zbase = (UnomB)^2/Snom

**Comment by ceraolo** [comment]
It depends on what side of the transformer you apply the Xcc transformed in ohm. If you use UNomB or UNomA, the impedance is to be applied at the B or A  side of the internal ideal transformer, respectively.

**Comment by AndreaBartolini** [comment]
All the transformers in PowerGrids (up to now) are implemented by the following PI-Network:

<img width="497" height="231" alt="Image" src="https://github.com/user-attachments/assets/5d26f5c1-849b-45ac-ac22-1d0a344cea08" />

in which:
- Y = 1/(R+jX) where R and X are the current parameters in ohm
- YA is null
- YB = G+JB where G and B are the current parameters in siemens

I suppose that Xcc is equivalent to X in the PI-Network, so the new parameters in p.u. should be:
- RccPu
- XccPu
- GPu
- BPu

And the base for all the parameters in p.u. should be:
  Zbase = (UNomB)^2/SNom

because the entire PI-Network is on the side_B, after the transformer ratio k.

Please confirm that this proposal is according to the ticket request, so I can proceed with the implementation.

**Comment by ceraolo** [comment]
I confirm.
When we give data in PU they are values not specifically referred to a transformer side. The Ohm values, instead, are totally different depending on the side of the ideal transformer. 
According to the diagram, all parameters are internally referred to the right side of the internal transformer, side "At", which are to be converted using UNomB.

**Comment by AndreaBartolini** [comment]
Implemented by PR #193
